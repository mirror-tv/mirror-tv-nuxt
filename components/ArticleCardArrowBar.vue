<template functional>
  <a
    class="article-card-arrow-bar"
    :href="props.href"
    target="_blank"
    rel="noreferrer noopener"
  >
    <span class="top-wrapper">
      <span class="top-wrapper__bar-title-wrapper bar-title-wrapper">
        <span class="bar-title" v-text="props.barTitle" />
      </span>
      <span class="top-wrapper__arrow-icon arrow-icon" />
    </span>
    <span class="bottom-wrapper">
      <img v-lazy="props.articleImgURL" class="article-img" alt="article-img" />
      <span class="bottom-wrapper__info-wrapper info-wrapper">
        <span class="article-title" v-text="props.articleTitle" />
        <span
          class="article-date"
          v-text="$options.methods.formatDate(props.articleDate)"
        />
      </span>
    </span>
  </a>
</template>
c
<script>
import dayjs from 'dayjs'

export default {
  props: {
    href: {
      type: String,
      default: ''
    },
    barTitle: {
      type: String,
      default: ''
    },
    articleImgURL: {
      type: String,
      default: ''
    },
    articleTitle: {
      type: String,
      default: ''
    },
    articleDate: {
      type: Date,
      default: () => new Date()
    }
  },
  methods: {
    formatDate(date) {
      return dayjs(date).format('YYYY/MM/DD HH:mm')
    }
  }
}
</script>

<style lang="scss" scoped>
.article-card-arrow-bar {
  display: block;
  @include media-breakpoint-up(xl) {
    width: 180px;
  }
}

.top-wrapper {
  display: flex;
  width: 100%;
  &__bar-title-wrapper {
    flex: 1 1 auto;
    width: 0;
  }
}

.bar-title-wrapper {
  height: 20px;
  background-color: #ffcc01;
  padding: 0 10px;
  display: flex;
}

.bar-title {
  font-size: 14px;
  color: #003366;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
}

.arrow-icon {
  display: flex;
  justify-content: center;
  align-items: center;
  width: 20px;
  height: 20px;
  background-color: #003366;
  &:after {
    content: '';
    width: 10px;
    height: 10px;
    display: block;
    border-top: 1px solid white;
    border-right: 1px solid white;
    transform: rotate(45deg);
    position: relative;
    right: 2px;
  }
}

.bottom-wrapper {
  display: flex;
  @include media-breakpoint-up(xl) {
    flex-direction: column;
  }
}

.article-img {
  width: 46%;
  object-fit: cover;
  @include media-breakpoint-up(xl) {
    width: 100%;
  }
}

.info-wrapper {
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  align-items: flex-start;
  padding: 11px 0 0 20px;
  @include media-breakpoint-up(xl) {
    padding: 11px 0 0 0;
  }
}

.article-title {
  font-size: 16px;
  color: #4a4a4a;
  text-align: justify;
  word-wrap: break-word;
  -webkit-line-clamp: 3;
  display: -webkit-box;
  -webkit-box-orient: vertical;
  overflow: hidden;
}

.article-date {
  font-size: 14px;
  color: #9b9b9b;
  @include media-breakpoint-up(xl) {
    margin: 11px 0 0 0;
  }
}
</style>
