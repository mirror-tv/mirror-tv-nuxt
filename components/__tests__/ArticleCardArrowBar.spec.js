import { shallowMount } from '@vue/test-utils'
import ArticleCardArrowBar from '../ArticleCardArrowBar'

describe('props', function() {
  test('Should render text by "barTitle" props', function() {
    const barTitleMock = 'barTitleMock'
    const wrapper = shallowMount(ArticleCardArrowBar, {
      propsData: {
        barTitle: barTitleMock
      }
    })
    const barTitle = wrapper.find('.bar-title')
    expect(barTitle.text()).toBe(barTitleMock)
  })

  test('Should render text by "articleImgURL" props', async function() {
    expect.assertions(1)
    const articleImgURLMock = 'articleImgURLMock'
    const wrapper = await shallowMount(ArticleCardArrowBar, {
      propsData: {
        articleImgURL: articleImgURLMock
      }
    })
    const articleImg = wrapper.find('.article-img')
    expect(articleImg.attributes()['data-src']).toBe(articleImgURLMock)
  })

  test('Should render text by "articleTitle" props', function() {
    const articleTitleMock = 'articleTitleMock'
    const wrapper = shallowMount(ArticleCardArrowBar, {
      propsData: {
        articleTitle: articleTitleMock
      }
    })
    const articleTitle = wrapper.find('.article-title')
    expect(articleTitle.text()).toBe(articleTitleMock)
  })

  test('Should render text by "articleDate" props', function() {
    const articleDateMockString = '2020/01/02 03:04'
    const articleDateMock = new Date(2020, 0, 2, 3, 4)
    const wrapper = shallowMount(ArticleCardArrowBar, {
      propsData: {
        articleDate: articleDateMock
      }
    })
    const articleDate = wrapper.find('.article-date')
    expect(articleDate.text()).toBe(articleDateMockString)
  })

  test('Should render href by "href" props', function() {
    const hrefMock = 'https://www.google.com'
    const wrapper = shallowMount(ArticleCardArrowBar, {
      propsData: {
        href: hrefMock
      }
    })
    expect(wrapper.attributes().href).toBe(hrefMock)
  })
})

test('snapshot test', async function() {
  const wrapper = await shallowMount(ArticleCardArrowBar, {
    propsData: {
      href: 'https://www.mirrormedia.mg/story/20200715edi026/',
      barTitle: '生活',
      articleImgURL:
        'https://www.mirrormedia.com.tw/assets/images/20200715135810-51b41bed253ae37fe37c5f9972844e09-mobile.jpg',
      articleTitle: '振興券上路最大咖好禮　黃偉哲：來台南旅遊消費抽房子',
      articleDate: new Date(2020, 0, 2, 3, 4, 5)
    }
  })
  expect(wrapper.element).toMatchSnapshot()
})
