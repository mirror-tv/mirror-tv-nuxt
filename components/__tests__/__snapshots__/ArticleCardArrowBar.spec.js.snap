// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test 1`] = `
<a
  class="article-card-arrow-bar"
  href="https://www.mirrormedia.mg/story/20200715edi026/"
  rel="noreferrer noopener"
  target="_blank"
>
  <span
    class="top-wrapper"
  >
    <span
      class="top-wrapper__bar-title-wrapper bar-title-wrapper"
    >
      <span
        class="bar-title"
      >
        生活
      </span>
    </span>
     
    <span
      class="top-wrapper__arrow-icon arrow-icon"
    />
  </span>
   
  <span
    class="bottom-wrapper"
  >
    <img
      alt="article-img"
      class="article-img"
      data-src="https://www.mirrormedia.com.tw/assets/images/20200715135810-51b41bed253ae37fe37c5f9972844e09-mobile.jpg"
      lazy="loading"
      src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
    />
     
    <span
      class="bottom-wrapper__info-wrapper info-wrapper"
    >
      <span
        class="article-title"
      >
        振興券上路最大咖好禮　黃偉哲：來台南旅遊消費抽房子
      </span>
       
      <span
        class="article-date"
      >
        2020/01/02 03:04
      </span>
    </span>
  </span>
</a>
`;
